File: app/src/main/java/com/arman/queuetube/util/notifications/receivers/WifiReceiver.java
package com.arman.queuetube.util.notifications.receivers;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.view.View;

import com.arman.queuetube.R;

public class WifiReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        try {
            Activity activity = (Activity) context;
            if (isOnline(context)) {
                activity.findViewById(R.id.error_message_container).setVisibility(View.GONE);
                activity.findViewById(R.id.ad_view).setVisibility(View.VISIBLE);
            } else {
                activity.findViewById(R.id.ad_view).setVisibility(View.GONE);
                activity.findViewById(R.id.error_message_container).setVisibility(View.VISIBLE);
            }
        } catch (Exception e) {
        }
    }

    private boolean isOnline(Context context) {
        ConnectivityManager manager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo info = manager.getActiveNetworkInfo();
        return info != null && info.isConnected();
    }

}

File: app/src/main/java/com/arman/queuetube/util/notifications/receivers/WifiReceiver.kt
package com.arman.queuetube.util.notifications.receivers

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.view.View
import com.arman.queuetube.R
import com.google.android.material.snackbar.Snackbar

class WifiReceiver : BroadcastReceiver() {

    private var snackbar: Snackbar? = null

    override fun onReceive(context: Context, intent: Intent) {
        try {
            val activity = context as Activity
            val container = activity.findViewById<View>(R.id.container)
            if (isOnline(context)) {
                this.snackbar?.dismiss()
            } else {
                this.snackbar = Snackbar.make(container, "You don't have an internet connection", Snackbar.LENGTH_INDEFINITE)
                this.snackbar!!.show()
            }
        } catch (e: Exception) {
        }

    }

    private fun isOnline(context: Context): Boolean {
        val manager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val info = manager.activeNetworkInfo
        return info != null && info.isConnected
    }

}
